app/src/main/java/fr/free/nrw/commons/navtab/MoreBottomSheetFragment.kt

```diff
@@ -182,16 +215,36 @@ class MoreBottomSheetFragment : BottomSheetDialogFragment() {
         Single.defer { single }
             .subscribeOn(Schedulers.io())
             .observeOn(AndroidSchedulers.mainThread())
-            .subscribe { success ->
+            .subscribe({ success ->
                 val messageResId = if (success) {
                     R.string.thanks_feedback
                 } else {
                     R.string.error_feedback
                 }
                 Toast.makeText(requireContext(), getString(messageResId), Toast.LENGTH_SHORT).show()
-            }
+            }, { error ->
+                error.printStackTrace()
+                Toast.makeText(requireContext(), R.string.error_feedback, Toast.LENGTH_SHORT).show()
+            })
+    }
+
+    /**
+     * This method is to check whether internet connection is available or not
+     */
+    fun isInternetConnectionAvailable(context: Context): Boolean {
+        val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            val activeNetwork: Network? = connectivityManager.activeNetwork
+            val networkCapabilities = connectivityManager.getNetworkCapabilities(activeNetwork)
+            networkCapabilities?.hasCapability(android.net.NetworkCapabilities.NET_CAPABILITY_INTERNET) == true
+        } else {
+            val activeNetworkInfo: NetworkInfo? = connectivityManager.activeNetworkInfo
```

Comment:
'NetworkInfo' is deprecated. Deprecated in Java
'NetworkInfo' is deprecated. Deprecated in Java
'getter for activeNetworkInfo: NetworkInfo?' is deprecated. Deprecated in Java
'getter for isConnected: Boolean' is deprecated. Deprecated in Java

Reply:
I had to use because the current minimum API level was 21 and I had to use it for devices below API level 23

Reply:
I don't think that checking the connection is the way to go, because:
1. It is complex.
2. Internet connection can be lost immediately after the apps checks it.

Instead I would recommend a try/catch approach:
1. Try to submit the feedback
2. If there was any kind of exception (might be internet connection error or server timeout or other issues), alert the user rather than crashing.

Reply:
check this @angrezichatterbox 
```
try{
//for example, retrofit call
}
catch (e: Exception) {
        is UnknownHostException -> "Unknown host!"
        is ConnectException -> "No internet!"
        else -> "Unknown exception!"
    }
```

Ref : https://stackoverflow.com/a/61099415